Makefile

```diff
@@ -25,10 +25,13 @@ AD_SERVER_DIR:=images/ad-server
 CLIENT_DIR:=images/client
 TOOLBOX_DIR:=images/toolbox
 SERVER_SRC_FILE:=$(SERVER_DIR)/Containerfile.fedora
+OPENSUSE_SERVER_SRC_FILE:=$(SERVER_DIR)/Containerfile.opensuse
```

Comment:
It is better not to have vendor specific logic within top-level `Makefile`, or else we may end with bloated and hard-to-maintain Makefile, where every distro has its own settings. 

IMHO, a better approach would be one of (or combination of) the following options:
- Use make's optional-assignment. Something like: `SERVER_SRC_FILE?=$(SERVER_DIR)/...` so the user may define it from make's invocations.
- Use top level `-include` which allows overriding variables. Something like the method we use in samba-operator for developer's mode [(here](https://github.com/samba-in-kubernetes/samba-operator/blob/master/Makefile#L2) and [build-from-source](https://github.com/samba-in-kubernetes/samba-operator/blob/master/docs/developers-notes.md#build-from-source) using [devel.mk](https://github.com/samba-in-kubernetes/samba-operator/blob/master/devel.mk.sample)

Reply:
That works for me. Hopefully that's good for @synarete too :-)

Reply:
This would be a valid option. 
However, I wonder if the order of `include` derivatives in `Makefile` would affect the variable evaluation within each make rule? I am not a `make` expert so need to play with this idea. If it works, it would be great solution.

Reply:
Cool!

Reply:
I've updated the MR.

Reply:
Looks cleaner and better. 

However: when I try to build the `opensuse` container on my `fedora36` machine I get the following errors:
```
No update candidate for 'glibc-2.36-9.1.x86_64'. The highest available version is already installed.
'python3-jsonschema' not found in package names. Trying capabilities.
'python3-pip' not found in package names. Trying capabilities.
'python3-pyxattr' not found in package names. Trying capabilities.
'sambacc' not found in package names. Trying capabilities.
No provider of 'sambacc' found.
```

(Sooner or later we would may want to add this build as part of CI).

Reply:
Fair enough. Assuming same holds for the python packages (`python3-jsonschem` `python3-pip` and `python3-pyxattr`).

Reply:
Actually, those aren't a fatal error for the python packages. It's just saying it's resolving them from python3-jsonschem to python39-jsonschem, for example. The fatal error is 'No provider of 'sambacc' found.'. You would see the same error for the python packages if those failed to resolve.